### ISSUE TITLE
[Suggestion] Better explain how to set up the server for multiple people to play

### ISSUE THREAD

**Issue by MajickTek** [yuwui/Uchu#312]
Both me and my friend tried setting up the server with the auth/character ports forwarded and were unable to join each other.

We even tried vpns such as Hamachi.

Someone on the discord told me i needed to set up a host name and pfx certificate so I did and it made zero difference

**Comment by teccheck** [comment]
Yeah, the problem is (at the moment) that you need a certificate from an official CA (a company that signs certificates). But they only sign certificates for domains on the Internet and so if your device is only in a local LAN this won't work.

We're working on a solution for this problem but It might take just a little bit of time

**Comment by MajickTek** [comment]
I assumed if I bought a domain with SSL from Namecheap it would work but it seems like it won't so it's unclear to me what this "official CA" stuff is about

**Comment by teccheck** [comment]
I think that should work, but I'm not really an expert on SSL/TLS and certs. Do you know who signed this certificate?

**Comment by MajickTek** [comment]
It doesn't i bought the SSL through namecheap and created the cert/pfx using the openssl tool.

maybe theres more to it than i think

EDIT: also the domain just has an "A record" pointing to my IP right now

**Comment by teccheck** [comment]
You can use Let's Encrypt if you have a domain: https://letsencrypt.org/

**Comment by enteryournamehere** [comment]
> It doesn't i bought the SSL through namecheap and created the cert/pfx using the openssl tool.

That should be fine. 
Could you copy the client logs here (or on Discord)? (they should appear if you launch it from the terminal)

**Comment by MajickTek** [comment]
Update: Still no luck also there are no logs to post, no errors or anything just the standard boot messages

**Comment by Jettford** [comment]
The Namecheap SSL cert will be signed by a valid certificate authority there is no question on whether it is a valid SSL cert. You need to be sure that the server is loading the SSL certificate and using it for its connections.

Please check if there is a log that starts with
`PRIVATE KEY:`

**Comment by Jettford** [comment]
It should be literally the first log the server outputs

**Comment by ArkShocer** [comment]
FYI - The user was able to solve his problems with the following actions:

**1.** Open all required ports
**2.** Generate certificate without password
**3.** Make all dlls in the lib folder executable (**chmod a+x**)

**Comment by MajickTek** [comment]
Although technically I solved this issue, in the end I was still likely missing something: The server can't seem to keep the client authentication when loading the rocket world after creating a new character.

With DLU releasing in about a week I don't expect this to be fixed right away but it would still be neat

**Comment by Jettford** [comment]
> Although technically I solved this issue, in the end I was still likely missing something: The server can't seem to keep the client authentication when loading the rocket world after creating a new character.
> 
> With DLU releasing in about a week I don't expect this to be fixed right away but it would still be neat

This is almost certainly a configuration issue not a server issue.

**Comment by MajickTek** [comment]
maybe a  config issue with my webserver but not with uchu itself. i copied the config files to my local pc and i can connect to it fine from localhost.

it's just on the webserver that it wont work
